chore: update contributing.md (#1206)
delta-io/delta-kernel-rs: CONTRIBUTING.md

# Contributing to Delta Kernel (Rust)

> [!NOTE]
> **Found a bug?** Please first search [existing issues] to avoid duplicates. If you find a related
> issue, add your details there. Otherwise, open a new issue with a reproducible example including
> Rust version, delta-kernel-rs version, code executed, and error message.

> [existing issues]: (https://github.com/delta-io/delta-kernel-rs/issues)

Author: oh just from a markdown perspective? (considering they render the same right?)

## How to Contribute

We follow a structured contribution process to ensure high-quality code:

1. **Start with an issue and/or design sketch**: Open an issue describing what you want to
   contribute and why. Continue to step 2 after reaching some consensus. This helps us avoid wasted
   effort (perhaps you were building something that someone else was already pursuing or already
   explored and rejected). Including a design sketch will help drive consensus (often a simple
   diagram or bullet points outlining high-level changes is sufficient).
2. **Prototype/POC**: Create a PR marked as prototype/draft (not intended to merge) and gather
   feedback to further derisk the design. This PR is not intended to be merged but will guide the
   implementation and serve as a proving ground for the design. Then, pieces are torn out into

>    smaller PRs that can be merged.

Reviewer: Let's give em something to target :)
```suggestion
   smaller PRs (1-200 LOC) that can be merged.
```

Author: hm fair point though I hesitate to put numbers in here since "small" for some things might be 600 lines (100 lines material change + 500 lines tests)

how about I merge this and we can take this up as an improvement if others feel it's useful?

3. **Implementation**: Finally, create PR(s) to implement the feature (production code, tests,
   thorough docs, etc.). Often the initial POC will be split into multiple smaller PRs (e.g.,
   refactors, then feature additions, then public APIs specifically). Care should be taken to ensure

>    each PR is easily reviewable and thoroughly (independently) tested.

Reviewer: nit: Not sure what independently tested means, and unsure if it adds much.
```suggestion
   each PR is easily reviewable and thoroughly tested.
```

## Forking and Setup

1. Fork the repository into your account
2. Clone your fork locally:
   ```bash
   git clone git@github.com:YOUR_USERNAME/delta-kernel-rs.git
   cd delta-kernel-rs
   ```
3. Add the upstream remote:
   ```bash
   git remote add upstream git@github.com:delta-io/delta-kernel-rs.git
   ```

Now you have:
- `origin` pointing to your fork
- `upstream` pointing to the original repository

## Development Workflow

Our trunk branch is named `main`. Here's the typical workflow:

1. Pull the latest main to get a fresh checkout:
   ```bash
   git checkout main
   git pull upstream main
   ```
2. Create a new feature branch:
   ```bash
   git checkout -b my-feature
   ```
   (NB: Consider using `git worktrees` for managing multiple branches!)

> 3. Make your changes and test them (cargo test, run clippy, etc. - see what our CI runs)

Reviewer: Let's include the actual commands 
```suggestion
3. Make your changes and test them using the following commands. See our CI runs for a full set of tests.
    ```rust
     cargo test --workspace --verbose --all-features -- --skip read_table_version_hdfs
     cargo clippy --benches --tests --all-features -- -D warnings`
      cargo doc --workspace --all-features
    ```
```
